packages/react/src/components/Grid/Grid.js

```diff
@@ -0,0 +1,311 @@
+/**
+ * Copyright IBM Corp. 2019
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React from 'react';
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+
+import { settings } from 'carbon-components';
+import { breakpoints } from '@carbon/layout';
+
+import { range } from '../../tools/array';
+
+const { prefix } = settings;
+
+// getter functions needed due to error in storybook
+// addons knobs that requires a copy of this object to avoid
+const getValidColWidths = () => ({
+  sm: [false, true, 'auto', ...range({ end: breakpoints.sm.columns })],
+  md: [false, true, 'auto', ...range({ end: breakpoints.md.columns })],
+  lgPlus: [false, true, 'auto', ...range({ end: breakpoints.lg.columns })],
+});
+
+const getValidColOffsets = () => ({
+  sm: range({ end: breakpoints.sm.columns - 1 }),
+  md: range({ end: breakpoints.md.columns - 1 }),
+  lgPlus: range({ end: breakpoints.lg.columns - 1 }),
+});
+
+const VALID_COL_WIDTHS = getValidColWidths();
+const VALID_COL_OFFSETS = getValidColOffsets();
+
+export const Grid = ({
```

Comment:
Quick style nits:

```jsx
export function Grid({
  as: BaseComponent = 'div',
  condensed = false
  fullWidth = false,
  className: containerClassName,
  ...rest,
}) {
  const className = cx(containerClassName, {
    [`${prefix}--grid`]: true,
    [`${prefix}--grid--condensed`]: condensed,
    [`${prefix}--grid--full-width`]: fullWidth,
  });
  return <BaseComponent className={className} {...rest} />;
}
```

Reply:
honestly, looking back at it, this syntax is easier to understand than what I wrote, not sure why I felt the need to change it... 🤷‍♂
279bc1d